### PR TITLE
Profile tool: Fix reporting app contains Dataset

### PR DESCRIPTION
**Pull request by tgravescs**
Found a bug where we aren't reporting the plan contains dataset properly. The issue here is timing when we set it and when we discover the dataset, which happens later. 

Signed-off-by: Thomas Graves <tgraves@nvidia.com>

### CODE REVIEW

**Comment by tgravescs** [comment]
build

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala [hunk 1]
```diff
@@ -233,7 +233,6 @@ class ApplicationInfo(
   val accumIdToStageId: mutable.HashMap[Long, Int] = new mutable.HashMap[Long, Int]()
   var taskEnd: ArrayBuffer[TaskCase] = ArrayBuffer[TaskCase]()
   var unsupportedSQLplan: ArrayBuffer[UnsupportedSQLPlan] = ArrayBuffer[UnsupportedSQLPlan]()
-  var datasetSQL: ArrayBuffer[DatasetSQLCase] = ArrayBuffer[DatasetSQLCase]()
```
Comment (revans2) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala:236]: Shouldn't the case class `DatasetSQLCase` also be removed? It looks like it is only used here.

Comment (tgravescs) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala:236]: yes it should thanks for catching

**Comment by tgravescs** [comment]
build

**Review by revans2 — APPROVED** [review]

**Review by nartal1 — APPROVED** [review]

**Pull request merged**